proposals/nnnn-trailing-comma-lists.md

```diff
@@ -0,0 +1,294 @@
+# Allow trailing comma in comma-separated lists
+
+- Proposal: SE-NNNN
+- Author: Mateus Rodrigues
+- Review Manager: TBD
+- Status: Awaiting Review
+- Implementation: [swift](https://github.com/mateusrodriguesxyz/swift/commit/114b3b4ba1440d6f681b7b5f1086633b504eb2fb), [swift-syntax](https://github.com/mateusrodriguesxyz/swift-syntax/commit/17d3123f1868f450c878cb017403c4c686e8fc86)
+- Review: [pitch](https://forums.swift.org/t/pitch-allow-trailing-comma-in-tuples-arguments-and-if-guard-while-conditions/70170/48)
+
+## Introduction
+
+This proposal aims to allow the use of trailing commas, currently restricted to array and dictionary literals, in more comma-separated lists.
+
+## Motivation
+
+### Development Quality of Life Improvement
+
+A trailing comma is an optional comma after the last item in a list of elements:
+
+```swift
+let rank = [
+  "Player 1",
+  "Player 3",
+  "Player 2",
+]
+```
+
+Using trailing commas makes it easy to add, remove, reorder and comment in/out elements, without the need to add or delete the comma while doing any of these manipulations.
```

Comment:
```suggestion
Swift's support for trailing commas in array and dictionary literals makes it as easy to append, remove, reorder, or comment out the last element as any other element.
```

To be precise, the benefits have to do with the _last_ element of a list, not other elements. So perhaps something like this? Also, it is a bit pedantic, but one can try to _use_ a trailing comma but if it's not supported then it's not easy...